geopandas/geodataframe.py

```diff
@@ -609,7 +609,44 @@ def to_parquet(self, filename, compression="snappy", index=None, **kwargs):
 
         from geopandas.io.parquet import _to_parquet
 
-        _to_parquet(self, filename, compression=compression, index=index, **kwargs)
+        _to_parquet(self, path, compression=compression, index=index, **kwargs)
+
+    def to_feather(self, path, index=None, **kwargs):
+        """Write a GeoDataFrame to the Feather format.
+
+        Any geometry columns present are serialized to WKB format in the file.
+
+        Requires 'pyarrow' >= 0.17.
+
+        WARNING: this is an initial implementation of Feather file support and
+        associated metadata.  This is tracking version 0.1.0 of the metadata
+        specification at:
+        https://github.com/geopandas/geo-arrow-spec
+
+        This metadata specification does not yet make stability promises.  As such,
+        we do not yet recommend using this in a production setting unless you are
+        able to rewrite your Feather files.
+
+        .. versionadded:: 0.8
+
+        Parameters
+        ----------
+        path : str, path object
+        compression : {'snappy', 'gzip', 'brotli', None}, default 'snappy'
```

Comment:
The compression options don't match the spec for Feather, if I'm reading it correctly:
https://arrow.apache.org/docs/python/generated/pyarrow.feather.write_feather.html

```
'zstd', 'lz4', 'uncompressed'
```

It looks like we specifically have to map `None` here to `uncompressed` for output, as the default behavior for `write_feather` is `lz4` if it is available.

Reply:
I would leave the default to match with pyarrow.feather (the default of None means: use lz4 if available, otherwise uncompressed). Updated the docstring for that.

Reply:
It looks like the docstring got updated for `_to_feather` but not `to_feather` also.

Having the compression defaults between Parquet / Feather bugs me a little bit from an API consistency standpoint where `compression=None` semantically reads as no compression, rather than use the default of the underlying lib.

But!  The performance for Feather + LZ4 looks _so_ much better that it's hard to argue against that as a sensible default.

Would it be too ugly to add `"uncompressed"` as an option for Parquet, where `compression=None` means "use the default" (which happens to be uncompressed) and `compression="uncompressed"` means explicitly do not compress it?

Reply:
@jorisvandenbossche it might be github hiding things from me, but it looks like this docstring for `to_feather` at the top level needs to be updated to match `_to_feather`.

Reply:
> docstring for to_feather at the top level needs to be updated to match _to_feather.

Yes, good catch, fixed.

Regarding the inconsistency about None vs "uncompressed", that's a good point. But I would rather like to bring this up with pyarrow to fix it there.

Of course we could easily add "uncompressed" to `to_parquet` here as well (but it would also be inconsistent with pandas, unless we add it there as well ..)

Reply:
👍 on taking this up in `pyarrow` instead.